### PR TITLE
[ws-manager-mk2] Test automation for ws-daemon

### PR DESCRIPTION
**Pull request by Furisto**
## Description
Test automation for ws-daemon

## Related Issue(s)

Fixes WKS-13

#### Preview status

gitpod:summary

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/ws-daemon/pkg/controller/workspace_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+		g.Expect(ws.Status.GitStatus.UncommitedFiles).To(Equal(gitStatus.UncommitedFiles))
+		g.Expect(ws.Status.GitStatus.TotalUncommitedFiles).To(Equal(gitStatus.TotalUncommitedFiles))
+		g.Expect(ws.Status.GitStatus.UntrackedFiles).To(Equal(gitStatus.UntrackedFiles))
+		g.Expect(ws.Status.GitStatus.TotalUntrackedFiles).To(Equal(gitStatus.TotalUntrackedFiles))
+		g.Expect(ws.Status.GitStatus.UnpushedCommits).To(Equal(gitStatus.UnpushedCommits))
+		g.Expect(ws.Status.GitStatus.TotalUnpushedCommits).To(Equal(gitStatus.TotalUnpushedCommits))
+	}, timeout, interval).Should(Succeed())
+}
+
+func markContentReady(ws *workspacev1.Workspace) {
```
Comment (WVerlaek) [inline components/ws-daemon/pkg/controller/workspace_controller_test.go:256]:
```suggestion
func markContentReadyAndStop(ws *workspacev1.Workspace) {
```

File: components/ws-daemon/pkg/controller/workspace_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+			ops.EXPECT().BackupWorkspace(gomock.Any(), gomock.Any()).Return(nil, fmt.Errorf("BOOM!")).Times(1)
+			ops.EXPECT().DeleteWorkspace(gomock.Any(), gomock.Any())
+			workspaceCtrl.operations = ops
+
+			_ = createSecret(fmt.Sprintf("%s-tokens", name), secretsNamespace)
+			ws := newWorkspace(name, workspaceNamespace, workspacev1.WorkspacePhaseCreating)
+			createWorkspace(ws)
+			markContentReady(ws)
+
+			expectConditionEventually(ws, string(workspacev1.WorkspaceConditionBackupFailure), metav1.ConditionTrue, "BackupFailed")
```
Comment (WVerlaek) [inline components/ws-daemon/pkg/controller/workspace_controller_test.go:106]: maybe nice to check here that the condition message contains the failure reason as well

**Review by WVerlaek — APPROVED** [review]
lgtm, two nits

**Comment by kylos101** [comment]
```
[components/ws-daemon:lib] FAIL! -- A BeforeSuite node failed so all tests were skipped.
[components/ws-daemon:lib] --- FAIL: TestAPIs (8.61s)
```

@Furisto @WVerlaek the build is failed :point_up: Any leads why? :thinking:

**Comment by WVerlaek** [comment]
@kylos101 yes see [earlier comment](https://github.com/gitpod-io/gitpod/pull/17502#issuecomment-1534973123)

**Pull request merged**